### PR TITLE
Allow parsing numbers as strings and vice versa in rest requests

### PR DESCRIPTION
**Pull request by volllly**

### CODE REVIEW

File: src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace fiskaltrust.Launcher.Helpers
+{
+    public class NumberToStringConverter : JsonConverter<string>
+    {
+        public override bool CanConvert(Type typeToConvert)
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs:9]: Just to be sure: `typeToConvert` is the target type?

File: src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                }
+            }
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                return reader.GetString();
+            }
+
+            using JsonDocument document = JsonDocument.ParseValue(ref reader);
+            return document.RootElement.Clone().ToString();
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs:34]: Why is it required to clone the element here?

File: src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace fiskaltrust.Launcher.Helpers
+{
+    public class NumberToStringConverter : JsonConverter<string>
+    {
+        public override bool CanConvert(Type typeToConvert)
+        {
+            return typeof(string) == typeToConvert;
+        }
+        public override string? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs:13]:
```suggestion

        public override string? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

File: src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs (hunk 2 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs:34]: And should we even support this case? What else could it be than a string or a number? 🤔

File: src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs (hunk 1 above)
Comment (volllly) [reply src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs]: correct, we can actually remove this override as thats just the default

File: src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs (hunk 2 above)
Comment (volllly) [reply src/fiskaltrust.Launcher/Helpers/NumberToStringConverter.cs]: fair point. If we throw on that case the collection still runs through so we can remove it 👌

**Review by TSchmiedlechner — APPROVED** [review]

**Pull request merged**